### PR TITLE
fix atomic::op::Cas

### PR DESCRIPTION
**Pull request by BenjaminW3**
Fixes issue #187 and additionally adds a test for all atomic operations with different types (if supported by the accelerator).

### CODE REVIEW

**Comment by mxmlnkn** [comment]
> EDIT: Has atomic::op::Cas ever worked on CUDA? It does not look like it should have worked on any accelerator at all.

I couldn't test on CUDA because of #186, I just assumed it would work, because the code was there

I did a `git-cherry-pick` on your commit, because `cupla` doesn't work with alpaka HEAD yet, and I get the following error:
    
    imresh/cupla/alpaka/include/alpaka/atomic/Traits.hpp:115:26: error: incomplete type 'alpaka::atomic::traits::AtomicOp<alpaka::atomic::op::Cas, const alpaka::acc::AccCpuOmp2Threads<std::integral_constant<long unsigned int, 3ul>, unsigned int>&, int, void>' used in nested name specifier
                         value);
                              ^
The only merge conflicts were in `.travis.yml` and `test/unit/atomic/CMakeLists.txt` so I don't think I changed anything in a wrong way. See this commit https://github.com/mxmlnkn/alpaka/commit/1ba560329cdbc30ca581d0c403ff841f5927571e

Note that I had the exact same bug when I tried to implement this fix myself, but I deleted my CMakeCache and everything.

**Comment by BenjaminW3** [comment]
Yes, I know. The CI tells me the same. I am trying to fix it...

**Comment by psychocoderHPC** [comment]
@BenjaminW3 thanks for the fast fixes. 

Please wait that the tests are finished before merging! Do not merge your own pull request this breaks the pear reviewing.

File: include/alpaka/atomic/Traits.hpp [hunk 1]
```diff
@@ -80,6 +82,41 @@ namespace alpaka
...
+        //! \param addr The value to change atomically.
+        //! \param compare The comparison value used in the atomic operation.
+        //! \param value The value used in the atomic operation.
+        //-----------------------------------------------------------------------------
+        ALPAKA_NO_HOST_ACC_WARNING
+        template<
+            typename TOp,
+            typename TAtomic,
+            typename T>
+        ALPAKA_FN_HOST_ACC auto atomicOp(
```
Comment (psychocoderHPC) [inline include/alpaka/atomic/Traits.hpp]: Why is this a fixed 4 elements signature and not something like.

```
ALPAKA_NO_HOST_ACC_WARNING
template<
    typename TOp,
    typename TAtomic,
    typename... TArgs>
ALPAKA_FN_HOST_ACC auto atomicOp(
    TAtomic const & atomic,
    TArgs & args...);
```

Comment (BenjaminW3) [inline include/alpaka/atomic/Traits.hpp]: There is no hard reason against your solution. The current version is just easier to debug if it is called with too many arguments as it shows the error at an earlier stage.

Comment (psychocoderHPC) [inline include/alpaka/atomic/Traits.hpp]: My point goes in the direction that not all atomics had a address, compare and value.
But I am fine with this implementation if this is still valid for all alpaka provided atomic.

**Pull request merged**